x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformTask.java

```diff
@@ -568,4 +576,45 @@ TransformTaskState getTaskState() {
         return context.getTaskState();
     }
 
+    public static PersistentTask<?> getTransformTask(String transformId, ClusterState clusterState) {
+        Collection<PersistentTask<?>> transformTasks = findTransformTasks(t -> t.getId().equals(transformId), clusterState);
+        if (transformTasks.isEmpty()) {
+            return null;
+        }
+        // Task ids are unique
+        assert (transformTasks.size() == 1);
+        PersistentTask<?> pTask = transformTasks.iterator().next();
+        if (pTask.getParams() instanceof TransformTaskParams) {
+            return pTask;
+        }
+        throw new ElasticsearchStatusException(
+            "Found transform persistent task [" + transformId + "] with incorrect params",
+            RestStatus.INTERNAL_SERVER_ERROR
+        );
```

Comment:
```suggestion
        throw new ElasticsearchStatusException(
            "Found transform persistent task [{}] with incorrect params",
            RestStatus.INTERNAL_SERVER_ERROR,
            transformId
        );
```

nit, not a big deal, just didn't know if you knew about the formatting provided in `ElasticsearchStatusException`